doc/ref/chap-osh-assign.md

```diff
@@ -26,11 +26,70 @@ This chapter describes OSH assignment, which looks like `x=value`.
 
 ### sh-append
 
-## Compound Data
+## Compound Assignment
+
+### sh-init-list
+
+Indexed and associative arrays may be initialized by an assignment of an
+initializer list.
+
+    arr=(1 2 3 4)
+    dict=([apple]=red [banana]=yellow [orange]=orange)
+
+An initializer list does *NOT* provide a new value that will be assigned to the
+LHS of the assignment.  The initializer list is rather considered *a set of
+instructions to modify the existing value of the LHS*.
+
+An initializer list has the following form: `(<words>...)`, where each word has
+one of the following forms:
+
+- `[<key>]=<value>` ... This assigns `<value>` to an element of the LHS
+  specified by `<key>`.  The `<value>` is not subject to word splitting and
+  pathname expansions as if it is the RHS of an assignment.
+- `[<key>]+=<value>` ... This appends `<value>` to an element of the LHS
+  specified by `<key>`.  If the corresponding element does not exist, it simply
+  assigns `<value>` to a new element associated with `<key>`.  The `<value>` is
+  not subject to word splitting and pathname expansions.
+- `<value>` ... If the word does not have the above two forms, it is considered
+  a normal word.  In this case, this assigns `<value>` to the *next* element,
+  where the next element is determined by the LHS.  Unlike the previous two
+  forms, the `<value>` is subject to word splitting and pathname expansions as
+  if it is a normal argument to a command.
+
+The above three forms can be mixed within one initializer list, though there
+may be additional limitations depending on the type of the LHS of the
+assignment.
+
+When the assignment is performed with `=`, the content of the LHS value is
```

Comment:
I would move this paragraph above after "a set of instructions to modify the existing value of the LHS"

Reply:
I mover the paragraph in 4c4f59333. Or do you think we should combine the two paragraphs into one?